client/finality-grandpa/src/import.rs

```diff
@@ -315,7 +320,7 @@ where
 		let is_descendent_of = is_descendent_of(&*self.inner, Some((hash, parent_hash)));
 
 		let mut guard = InnerGuard {
-			guard: Some(self.authority_set.inner().write()),
+			guard: Some(self.authority_set.inner_locked()),
```

Comment:
Technically we don't need to lock here as we won't be releasing the outer lock (there's no await boundary below).

Reply:
Yeah, not here, but in the outer context that is calling this function and that requires that the set stays locked.